embark/embark/consumers.py

```diff
@@ -0,0 +1,173 @@
+import json
+import logging
+
+import os
+import channels
+import rx
+from rx import Observable
+import rx.operators as ops
+import re
+from inotify_simple import flags
+from . import inotify_wrap
+import difflib
+from channels.generic.websocket import WebsocketConsumer
+from pathlib import Path
+
+
+# consumer class for synchronous/asynchronous websocket communication
+# TODO: Implement frontend Websocket handling and data extraction (Example in progress.html -> remove later)
+class WSConsumer(WebsocketConsumer):
+
+    # constructor
+    def __init__(self):
+        super().__init__()
+        # self.room_group_name = 'status_updates_group'
+        # global module count and status_msg directory
+        self.module_count = 0
+        # TODO: extend dictionary for more information
+        self.status_msg = {
+            "percentage": 0.0,
+            "module": "",
+            "phase": "",
+        }
+
+    # this method is executed when the connection to the frontend is established
+    def connect(self):
+        # accept socket connection
+        self.accept()
+
+        # if file does not exist create it otherwise delete its content
+        open('/app/emba/log_1/emba_new.log', 'w+')
+
+        # start waiting for events
+        self.read_loop()
+
+    # called when received data from frontend TODO: implement this for processing client input at backend
+    def receive(self, text_data=None, bytes_data=None):
+        pass
+
+    # called when websocket connection is closed TODO: implement connection close if necessary
+    def disconnect(self, close_code):
+        pass
+
+    # send data to frontend
+    def send_data(self):
+        self.send(json.dumps(self.status_msg, sort_keys=True))
+
+    # update our dict whenever a new module is being processed
+    def update_status(self, stream_item_list):
+        # progress percentage TODO: improve percentage calculation
+        self.module_count += 1
+        percentage = self.module_count / 35
+        self.status_msg.update({"module": stream_item_list[0]})
```

Comment:
`self.status_msg["module"] = stream_item_list[0]`

This will do what you want. `update` is an overkill in my opinion